lib/egon/overcloud/undercloud_handle.rb

```diff
@@ -39,6 +39,10 @@ def execute_workflow(workflow, input, wait_for_complete = true)
       workflow_execution_id
     end
 
+    def workflow_action_execution(action_name, params = {})
+      return JSON.parse(service('Workflow').create_action_execution(action_name, params).body["output"])["result"]
```

Comment:
Weird, I wonder why that didn't hit me.  I've added the requires, can you let me know if that fixes the issue for you?